web/src/utils/network.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) [2022-2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import ipaddr from "ipaddr.js";
+import {
+  APIRoute,
+  ApFlags,
+  ApSecurityFlags,
+  Connection,
+  ConnectionState,
+  Device,
+  IPAddress,
+  Route,
+  SecurityProtocols,
+} from "~/types/network";
+
+/**
+ * Returns a human readable connection state
+ */
+const connectionHumanState = (state: number): string => {
+  const stateIndex = Object.values(ConnectionState).indexOf(state);
+  const stateKey = Object.keys(ConnectionState)[stateIndex];
+  return stateKey.toLowerCase();
+};
+
+/**
+ * Check if an IP is valid
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An IP Address
+ * @return true if given IP is valid; false otherwise.
+ */
+const isValidIp = (value: IPAddress["address"]): boolean =>
+  ipaddr.IPv4.isValidFourPartDecimal(value);
+
+/**
+ * Check if a value is a valid netmask or network prefix
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An netmask or a network prefix
+ * @return true if given IP is valid; false otherwise.
+ */
+const isValidIpPrefix = (value: IPAddress["prefix"]): boolean => {
+  const prefix = value as string;
+
+  if (prefix.match(/^\d+$/)) {
+    return parseInt(prefix) <= 32;
+  } else {
+    return ipaddr.IPv4.isValidFourPartDecimal(prefix);
+  }
+};
+
+/**
+ * Prefix for the given prefix or netmask
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An netmask or a network prefix
+ * @return prefix for the given netmask or prefix
+ */
+const ipPrefixFor = (value: string): number => {
+  if (value.match(/^\d+$/)) {
+    return parseInt(value);
+  } else {
+    return ipaddr.IPv4.parse(value).prefixLengthFromSubnetMask();
+  }
+};
+
+/**
+ *  Converts an IP given in decimal format to text format
+ *
+ * FIXME: IPv6 is not supported yet.
```

Comment:
I think it is something that should deserve follow up card as ipv6 start being more common

Reply:
For sure! Adding it to the pending things in the PR description for easing the follow-up creation. Thanks!